lib/influx.js

```diff
@@ -0,0 +1,38 @@
+import zlib from 'node:zlib';
+import { promisify } from 'node:util';
+
+/*
+* Sends compressed request metrics to InfluxDB
+* https://docs.influxdata.com/enterprise_influxdb/v1/guides/write_data/
+*/
+export const reportMetric = async (request, response, env) => {
+  const compress = promisify(zlib.gzip);
+
+  // Define API endpoint and headers
+  const url = `${env.INFLUX_URL}/api/v2/write?&bucket=${env.INFLUX_DATABASE}&precision=ms`;
```

Comment:
1. Let's create the URL only once, outside of the function `reportMetric`. 
2. Let's build the querystring using the URLSearchParams object to correctly handle url-escaping of values like the bucket name.

```js
const INFLUX_SEARCH_PARAMS = new URLSearchParams({
  bucket: env.INFLUX_DATABASE,
  precision: 'ms',
})

const INFLUX_WRITE_URL = `${env.INFLUX_URL}/api/v2/write?${INFLUX_SEARCH_PARAMS}`

export const reportMetric = async (request, response, env) => {
 // ...
 return fetch(INFLUX_WRITE_URL, {
   // ...
 })
}
```

@juliangruber What do you think? Am I over-optimising a place that does not need that?

Reply:
> I don't know if this will be possible as `env` is only accessible as argument in context of cloudflare worker `fetch` function.
> 
> Here's an thread about that https://community.cloudflare.com/t/how-to-access-environment-variables-from-libraries/480041

Oh! I completely missed that env is provided in the worker context.

Please disregard my suggestion, I was wrong.